astro/terraform/terraform_remote_state_disable_utils.go

```diff
@@ -66,24 +69,51 @@ func deleteTerraformBackendConfig(in []byte) (updatedConfig []byte, err error) {
 		return nil, errors.New("could not parse \"terraform\" block in config")
 	}
 
-	if err := astDel(terraformConfigBlock.List, "backend"); err != nil {
-		return nil, err
-	}
+	astDelIfExists(terraformConfigBlock.List, "backend")
```

Comment:
How come the error handling is removed here?

Reply:
The logic of deleteTerraformBackendConfigWithHCL was: find terraform block, find and delete backend block from it, if backend block was not found, bubble up the error returned from astDel method. However there're valid terraform configurations which have terraform block without backend block inside (https://www.terraform.io/docs/configuration/terraform.html, for example required_version or required_provider). Previously astro plan --detach would fail on them.

I've changed astDel to astDelIfExists in this revision. astDelIfExists only deletes node from AST tree, if it was found there. It returns True if the item was removed and False otherwise, so there's no error handling to be done here. I can add a log tracing saying "no backend config was found, moving on". If we want to show an error, if no backend config was found at all, this needs to happen in deleteBackendConfig, where we're looping through the list of candidates (files that have "terraform" configuration in them).

Let me know, if you'd like to have log trace or an error handing one level above (which I think makes most sense).

Reply:
OK cool, makes sense! Thanks for the explanation.